File: app/src/main/java/by/d1makrat/library_fm/adapter/pages/TabTopArtistsAdapter.java
package by.d1makrat.library_fm.adapter.pages;

import android.net.Uri;
import android.os.Bundle;
import android.support.v4.app.FragmentManager;

import by.d1makrat.library_fm.AppContext;
import by.d1makrat.library_fm.model.Period;
import by.d1makrat.library_fm.ui.fragment.top.TopArtistsFragment;

import static by.d1makrat.library_fm.Constants.PERIOD_KEY;

public class TabTopArtistsAdapter extends TabTopAdapter {

    public TabTopArtistsAdapter(FragmentManager fragmentManager) {
        super(fragmentManager);
    }

    @Override
    public TopArtistsFragment getItem(int position) {
        Bundle bundle = new Bundle();
        bundle.putString(PERIOD_KEY, new Period().getValueForApi(position));
        TopArtistsFragment fragment = new TopArtistsFragment();
        fragment.setArguments(bundle);
        return fragment;
    }

    @Override
    public Uri getUrlForBrowser(int pPosition){
        return Uri.parse(AppContext.getInstance().getUser().getUrl() + "/library/artists" + new Period().getSuffixForUrl(pPosition));
    }
}

File: app/src/main/java/by/d1makrat/library_fm/adapter/pages/TabTopArtistsAdapter.kt
package by.d1makrat.library_fm.adapter.pages

import android.os.Bundle
import android.support.v4.app.FragmentManager
import by.d1makrat.library_fm.Constants.PERIOD_KEY
import by.d1makrat.library_fm.model.Period
import by.d1makrat.library_fm.ui.fragment.top.TopArtistsFragment

class TabTopArtistsAdapter(fragmentManager: FragmentManager) : TabTopAdapter(fragmentManager) {

    override fun getItem(position: Int): TopArtistsFragment {
        val bundle = Bundle()
        bundle.putString(PERIOD_KEY, Period().getValueForApi(position))
        val fragment = TopArtistsFragment()
        fragment.arguments = bundle
        return fragment
    }
}
